extensions/panache/hibernate-reactive-panache/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/PanacheEntityBase.java

```diff
@@ -58,6 +58,19 @@
         return INSTANCE.persist(this).map(v -> (T) this);
     }
 
+    /**
+     * Update this entity in the database, if not already persisted. This will set your ID field if it is not already set.
+     * In case of the entity is not attached to the session, this method will attach it to the current session.
+     *
+     * @return
+     *
+     * @see Mutiny.Session#merge(Object)
+     */
+    @CheckReturnValue
+    public <T extends PanacheEntityBase> Uni<T> merge() {
```

Comment:
There is a long history of demands for a `merge` method on Panache and this was always refused.

I think it's better to not open the debate again and do what we do in the imperative version: load then persist.

Reply:
Any hints about how to do the load and then persist here?

Let me give more details: the problem in this context is that the entity is created by the REST layer, so we have a method like:

```java
@POST
Response update(Entity entity) {
// ... 
}
```

The above method is auto generated and the entity is a detached object from the session. 

Alternatively, if your complain is that this method is exposed in `PanacheEntityBase`, I could do the same as in the Panache ORM Rest Data version:

```
@Transactional
   public Object update(Object var1, Object var2) {
      ((AbstractEntity)var2).setId((Number)var1);
      return EmptyListItem.getEntityManager().merge(var2);
   }
```

But without exposing this method in PanacheEntityBase.

Wdyt?

Reply:
Yes please, do the same as the Panache ORM version: don't expose the method on PanacheEntityBase and access the EntityManager/Mutiny Session directly.

Reply:
And please don't forget to do the same for the entity pattenr ;)

Reply:
Okis, doing it!

Reply:
Done